x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/configuration_step/analysis_fields_table.tsx

```diff
@@ -0,0 +1,190 @@
+  item.is_required === true;
+
+export const AnalysisFieldsTable: FC<{
+  excludes: string[];
+  loadingItems: boolean;
+  setFormState: any;
+  tableItems: FieldSelectionItem[];
+}> = ({ excludes, loadingItems, setFormState, tableItems }) => {
+  const [sortableProperties, setSortableProperties] = useState();
+
+  useEffect(() => {
+    let sortablePropertyItems = [];
+    const defaultSortProperty = 'name';
+
+    sortablePropertyItems = [
+      {
+        name: 'name',
+        getValue: (item: any) => item.name.toLowerCase(),
+        isAscending: true,
+      },
+      {
+        name: 'is_included',
+        getValue: (item: any) => item.is_included,
+        isAscending: true,
+      },
+      {
+        name: 'is_required',
+        getValue: (item: any) => item.is_required,
+        isAscending: true,
+      },
+    ];
+    const sortableProps = new SortableProperties(sortablePropertyItems, defaultSortProperty);
+
+    setSortableProperties(sortableProps);
+  }, []);
+
+  const filters = [
+    {
+      type: 'field_value_selection',
+      field: 'is_included',
+      name: i18n.translate('xpack.ml.dataframe.analytics.create.excludedFilterLabel', {
+        defaultMessage: 'Is included',
+      }),
+      multiSelect: false,
+      options: [
+        {
+          value: true,
+          view: (
+            <EuiText grow={false}>
+              {i18n.translate('xpack.ml.dataframe.analytics.create.isIncludedOption', {
+                defaultMessage: 'Yes',
+              })}
+            </EuiText>
+          ),
+        },
+        {
+          value: false,
+          view: (
+            <EuiText grow={false}>
+              {i18n.translate('xpack.ml.dataframe.analytics.create.isNotIncludedOption', {
+                defaultMessage: 'No',
+              })}
+            </EuiText>
+          ),
+        },
+      ],
+    },
+  ];
+
+  return (
+    <Fragment>
+      <EuiFormRow
+        label={i18n.translate('xpack.ml.dataframe.analytics.create.excludedFieldsLabel', {
+          defaultMessage: 'Excluded fields',
+        })}
+        helpText={i18n.translate(
+          'xpack.ml.dataframe.analytics.create.excludedFieldsLabelHelpText',
+          {
+            defaultMessage: 'From included fields, select fields to exclude from analysis.',
+          }
+        )}
+      >
+        <Fragment />
+      </EuiFormRow>
+      {tableItems.length === 0 && (
+        <EuiCallOut
+          title={i18n.translate('xpack.ml.dataframe.analytics.create.calloutTitle', {
+            defaultMessage: 'Analysis fields not available',
+          })}
+        >
+          <FormattedMessage
+            id="xpack.ml.dataframe.analytics.create.calloutMessage"
+            defaultMessage="Additional data required to load analysis fields."
+          />
+        </EuiCallOut>
+      )}
+      {tableItems.length > 0 && (
+        <EuiPanel paddingSize="m">
+          <CustomSelectionTable
```

Comment:
Assuming we stick with this table for selecting / deselecting fields, we should make sure the page number is retained when the table is reloaded following a change to the selection. Currently it is confusing how the page resets to `1` if you change the selection on e.g. page `2`

![image](https://user-images.githubusercontent.com/7405507/83403211-253e0c00-a400-11ea-92ed-3c8511decb9c.png)

Reply:
Fixed in 706f449459b3cc03f0bea7fe6483d423896e11b6

Reply:
I am still seeing the table go back to page 1 when selecting or deselecting items after the changes in https://github.com/elastic/kibana/commit/706f449459b3cc03f0bea7fe6483d423896e11b6.

Reply:
@peteharverson - I was able to use a custom comparison with react memo to ensure the table isn't re-rendered unnecessarily. Updated fix here cbdf3d5495501b77667dcb273df7817594782d25